### ISSUE TITLE
Added symbol node kind. [phab:D1733]

### ISSUE THREAD

**Issue by kythe** [google/kythe#2166]
_Created by **craigbarber** at 2017-06-23 18:16:42_: 

Modified the schema definition to include a new node kind "symbol".
Also modified the java analyzers shared enums to include the new kind.

**Comment by kythe** [comment]
_**danielmoy-google** commented inline (context lost during migration) at 2017-06-23 18:23:58_:

is there one extra newline in here?  (idk if we even care about that sort of thing)

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2017-06-23 18:24:16_:

```
Brief description::
  A *symbol* is a common name used by tools to refer to a set of objects.
  The rules for binding a symbol to a particular object from this set may depend on 
  external configuration (such as the list of libraries being linked together to produce
  an executable).
Naming convention::
  Language:::
    No special rules (though see below).
  Path:::
    *empty*
  Root:::
    *empty*
  Corpus:::
    *empty*
  Signature:::
    No special rules.

The languages `ia64` and `jvm` are reserved.
```

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2017-06-23 18:26:13_:

We don't care too much as long as asciidoc still works. :) But I try to keep the format consistent so that kythe/go/util/tools/extractschema will continue to work (though we can also fix the tool if we change conventions).

**Comment by kythe** [comment]
_**craigbarber** wrote at 2017-06-23 18:30:43_:

  - incorporated review feedback

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2017-06-23 18:45:21_:

zarko: Do we want to reserve these here, or as top-level language labels? I think the latter, but maybe you have something else in mind?

**Comment by kythe** [comment]
_**zarko** wrote at 2017-06-23 20:46:07_:

Good point--let's strike the "The languages `ia64` and `jvm` are reserved." line and deal with those languages in a different diff.

**Comment by kythe** [comment]
_**zarko** wrote at 2017-06-23 20:47:28_:

(accepted modulo removing the line)

**Comment by kythe** [comment]
_**zarko** accepted this revision at 2017-06-23 20:47:28_

**Comment by kythe** [comment]
_**craigbarber** wrote at 2017-06-23 20:51:36_:

  - incorporated review feedback

**Comment by kythe** [comment]
_**craigbarber** closed via commit 880c03d4425bed357cbc7642bcab0b2125754cfc at 2017-06-23 20:52:52_
